packages/components/src/Avatar/src/Avatar.tsx

```diff
@@ -0,0 +1,254 @@
+
+        return size === "xs" && letters.length > 1 ? letters.charAt(0) : letters;
+    }, [name, size]);
+
+    return (
+        <Text
+            size={AvatarToTextSizeAdapter[size]}
+            className={styles["hop-Avatar__initials"]}
+        >
+            {initials}
+        </Text>
+    );
+}
+
+function Avatar(props: AvatarProps, ref: ForwardedRef<HTMLDivElement>) {
+    [props, ref] = useContextProps(props, ref, AvatarContext);
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const {
+        "aria-label": ariaLabel,
+        className,
+        fallbackSrc,
+        imageProps,
+        isDisabled,
+        name,
+        size: sizeValue,
+        slot,
+        src,
+        style,
+        ...otherProps
+    } = ownProps;
+    
+    const { onError, ...otherImageProps } = imageProps ?? {};
+    const [imageUrl, handleImageError, imageFailed] = useImageFallback(src, fallbackSrc);
+
+    const size = useResponsiveValue(sizeValue) ?? "md";
+
+    let classNames = clsx(
+        className,
+        GlobalAvatarCssSelector,
+        cssModule(
+            styles,
+            "hop-Avatar",
+            size,
+            getColorName(name)
+        ),
+        stylingProps.className
+    );
+
+    const brokenImageClassNames = clsx(
+        className,
+        GlobalAvatarCssSelector,
+        cssModule(
+            styles,
+            "hop-Avatar",
+            "broken-image"
+        ),
+        stylingProps.className
+    );
+
+    const mergedStyles: CSSProperties = {
+        ...stylingProps.style,
+        ...style
+    };
+
+    const initialsContent = <AvatarInitials
+        {...otherProps}
+        name={name}
+        size={size}
+    />;
+
+    const commonProps = {
+        ...otherProps,
+        ariaLabel,
+        ref,
+        size,
+        style: mergedStyles
+    };
+
+    let content = initialsContent;
+    
+    if (src) {
+        if (imageFailed) {
+            if (fallbackSrc === undefined) {
+                return (
+                    <RichIconAvatarImage
+                        {...otherProps}
+                        {...commonProps}
+                        className={brokenImageClassNames}
+                        isDisabled={isDisabled}
+                        slot={slot}
+                    >
+                        <BrokenImageRichIcon />
+                    </RichIconAvatarImage>
+                );
+            }
+        } else {
+            classNames = clsx(
+                className,
+                GlobalAvatarCssSelector,
+                cssModule(
+                    styles,
+                    "hop-Avatar",
+                    "image",
+                    size
+                ),
+                stylingProps.className
+            );
+
+            content = <img
+                {...otherImageProps}
+                src={imageUrl}
+                alt=""
+                aria-hidden
+                onError={onError || handleImageError}
```

Comment:
If you add onError to the props, it should be addittive, not replacing the avatar's behavior. 
```
const handleError = (e) => {
    handleImageError(e);
    onError?.(e)
}
...
content = <img
    {...otherImageProps}
    src={imageUrl}
    alt=""
    aria-hidden
    onError={handleError }
```

Reply:
ya I wasn't completely sure on this behavior, thanks I'll do that.